go/vt/sqlparser/ast_format.go

```diff
@@ -2760,9 +2718,15 @@ func (node *Variable) Format(buf *TrackedBuffer) {
 	case VariableScope:
 		buf.literal("@")
 	case SessionScope:
+		if node.Name.EqualString("transaction_isolation") || node.Name.EqualString("transaction_read_only") {
+			buf.astPrintf(node, "@@%s.", node.Scope.ToString())
```

Comment:
Why do we need to print the session scope specifically for the transaction case?

Reply:
without `session` they have meaning for next transaction scope only. I will add a new scope to explicitly call that out.

Reply:
because `@@` is different than `@@session.` in transaction related variables.